app/src/main/java/io/homeassistant/companion/android/widgets/media_player_controls/MediaPlayerControlsWidget.kt

```diff
@@ -0,0 +1,479 @@
+        val appWidgetId = intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, -1)
+
+        Log.d(
+            TAG, "Broadcast received: " + System.lineSeparator() +
+                    "Broadcast action: " + action + System.lineSeparator() +
+                    "AppWidgetId: " + appWidgetId
+        )
+
+        ensureInjected(context)
+
+        mediaPlayCtrlWidgetDao = AppDatabase.getInstance(context).mediaPlayCtrlWidgetDao()
+
+        super.onReceive(context, intent)
+        when (action) {
+            RECEIVE_DATA -> saveEntityConfiguration(context, intent.extras, appWidgetId)
+            UPDATE_MEDIA_IMAGE -> updateAppWidget(context, appWidgetId)
+            CALL_PREV_TRACK -> callPreviousTrackService(context, appWidgetId)
+            CALL_REWIND -> callRewindService(context, appWidgetId)
+            CALL_PLAYPAUSE -> callPlayPauseService(context, appWidgetId)
+            CALL_FASTFORWARD -> callFastForwardService(context, appWidgetId)
+            CALL_NEXT_TRACK -> callNextTrackService(context, appWidgetId)
+        }
+    }
+
+    private fun saveEntityConfiguration(context: Context, extras: Bundle?, appWidgetId: Int) {
+        if (extras == null) return
+
+        val entitySelection: String? = extras.getString(EXTRA_ENTITY_ID)
+        val labelSelection: String? = extras.getString(EXTRA_LABEL)
+        val showSkip: Boolean? = extras.getBoolean(EXTRA_SHOW_SKIP)
+        val showSeek: Boolean? = extras.getBoolean(EXTRA_SHOW_SEEK)
+
+        if (entitySelection == null || showSkip == null || showSeek == null) {
+            Log.e(TAG, "Did not receive complete configuration data")
+            return
+        }
+
+        mainScope.launch {
+            Log.d(
+                TAG, "Saving service call config data:" + System.lineSeparator() +
+                "entity id: " + entitySelection + System.lineSeparator()
+            )
+            mediaPlayCtrlWidgetDao.add(
+                MediaPlayerControlsWidgetEntity(
+                    appWidgetId,
+                    entitySelection,
+                    labelSelection,
+                    showSkip,
+                    showSeek
+                )
+            )
+
+            onUpdate(context, AppWidgetManager.getInstance(context), intArrayOf(appWidgetId))
+        }
+    }
+
+    private fun callPreviousTrackService(context: Context, appWidgetId: Int) {
+        mainScope.launch {
+            Log.d(TAG, "Retrieving media player entity for app widget $appWidgetId")
+            val entity: MediaPlayerControlsWidgetEntity? = mediaPlayCtrlWidgetDao.get(appWidgetId)
+
+            if (entity == null) {
+                Log.d(TAG, "Failed to retrieve media player entity")
+                return@launch
+            } else {
+                Log.d(
+                    TAG, "Calling previous track service:" + System.lineSeparator() +
+                            "entity id: " + entity.entityId + System.lineSeparator()
+                )
+            }
```

Comment:
Don't need the else since you are returning in the if
```suggestion
            }
            Log.d(
                TAG, "Calling previous track service:" + System.lineSeparator() +
                        "entity id: " + entity.entityId + System.lineSeparator()
            )
```

Reply:
Changed in e3a66e5